backend/corpora/common/entities/tiledb_data.py

```diff
@@ -0,0 +1,416 @@
+import os
+import shutil
+import numpy as np
+import tiledb
+import uuid
+import ast
+import time
+
+from backend.corpora.dataset_processing.process import process
+
+"""
+SCHEMA WITH TILEDB VERSIONING:
+
+s3://single-cell-corpus/
+
+    metadata/ [GROUP]
+        collections [ARRAY]
+            - uuid
+            - owner
+            - visibility
+            - revision_of
+            - datasets
+            - other metadata
+        datasets [ARRAY]
+            - uuid
+            - name
+            - other metadata
+
+    datasets/<UUID>/
+        artifacts/
+        <UUID>.h5ad
+        <UUID>.rds
+        <UUID>.cxg
+    wmg/<SNAPSHOT>/
+    soma/<SNAPSHOT>/
+"""
+
+class Utils():
+    attrs = {
+        "collections": [
+            # API
+            "name",
+            "description",
+            "contact_name",
+            "contact_email",
+            "links",
+            "datasets",
+            "curator_name",
+            "created_at",
+            "updated_at",
+            "published_at",
+            "publisher_metadata",
+            # internal TileDB schema
+            "visibility",
+            "revision_of",
+            "owner",
+        ],
+        "datasets": [
+            "X_approximate_distribution",
+            "X_normalization_assay",
+            "cell_count",
+            "cell_type",
+            "development_stage",
+            "disease",
+            "ethnicity",
+            "is_primary_data",
+            "name",
+            "organism",
+            "sex",
+            "tissue",
+            "explorer_url"
+        ]
+    }
+       
+    attrs_to_parse = {
+        "collections":   [
+            "links",
+            "datasets",
+            "publisher_metadata"
+        ],
+        "datasets": [
+            "assay",
+            "cell_type",
+            "development_stage",
+            "disease",
+            "ethnicity",
+            "organism",
+            "sex",
+            "tissue"
+        ]
+    }
+  
+    @staticmethod
+    def new_id():
+        return uuid.uuid4().hex
+
+    # TODO: figure out actual list and dict support in TileDB
+    @staticmethod
+    def parse_stored_data(data: dict, array: str) -> dict:
```

Comment:
Why is this method required?

Reply:
I don't know how to use dict/object attributes in TileDB, so I turn them into a string for storage and parse it when needed.

Reply:
And why do you need to `eval` them?

Reply:
To convert the string into whatever data type it should be.